integration-tests/smoke/ccip/ccip_disable_lane_test.go

```diff
@@ -0,0 +1,137 @@
+
+	// add all lanes
+	testhelpers.AddLanesForAll(t, &tenv, state)
+
+	var (
+		chains                 = e.AllChainSelectors()
+		chainA, chainB, chainC = chains[0], chains[1], chains[2]
+		expectedSeqNumExec     = make(map[testhelpers.SourceDestPair][]uint64)
+		startBlocks            = make(map[uint64]*uint64)
+		pairs                  = testsetups.GetSourceDestPairs([]uint64{chainA, chainB, chainC})
+
+		sendmessage = func(src, dest uint64, deployer *bind.TransactOpts) (*onramp.OnRampCCIPMessageSent, error) {
+			return testhelpers.DoSendRequest(
+				t,
+				e,
+				state,
+				testhelpers.WithSender(deployer),
+				testhelpers.WithSourceChain(src),
+				testhelpers.WithDestChain(dest),
+				testhelpers.WithTestRouter(false),
+				testhelpers.WithEvm2AnyMessage(router.ClientEVM2AnyMessage{
+					Receiver:     common.LeftPadBytes(state.Chains[chainB].Receiver.Address().Bytes(), 32),
+					Data:         []byte("hello"),
+					TokenAmounts: nil,
+					FeeToken:     common.HexToAddress("0x0"),
+					ExtraArgs:    nil,
+				}))
+		}
+
+		assertSendRequestReverted = func(src, dest uint64, deployer *bind.TransactOpts) {
+			_, err = sendmessage(src, dest, deployer)
+			require.Error(t, err)
+			require.Contains(t, err.Error(), "execution reverted")
+		}
+
+		assertRequestSent = func(src, dest uint64, deployer *bind.TransactOpts) {
+			latestHeader, err := e.Chains[dest].Client.HeaderByNumber(testcontext.Get(t), nil)
+			require.NoError(t, err)
+			block := latestHeader.Number.Uint64()
+			messageSentEvent, err := sendmessage(src, dest, e.Chains[src].DeployerKey)
+			require.NoError(t, err)
+			expectedSeqNumExec[testhelpers.SourceDestPair{
+				SourceChainSelector: src,
+				DestChainSelector:   dest,
+			}] = []uint64{messageSentEvent.SequenceNumber}
+			startBlocks[dest] = &block
+		}
+	)
+
+	// disable lane A -> B
+	testhelpers.RemoveLane(t, &tenv, chainA, chainB, false)
+	// send a message to confirm it is reverted between chainA and chainB
+	assertSendRequestReverted(chainA, chainB, e.Chains[chainA].Users[0])
+	// send a message between chainB and chainA to confirm it is not reverted
+	assertRequestSent(chainB, chainA, e.Chains[chainB].Users[0])
+	// disable lane B -> A
+	testhelpers.RemoveLane(t, &tenv, chainB, chainA, false)
+	assertSendRequestReverted(chainB, chainA, e.Chains[chainB].Users[0])
+
+	// send message in other lanes and ensure they are delivered
+	go func() {
+		assertRequestSent(chainA, chainC, e.Chains[chainA].Users[1])
+		assertRequestSent(chainC, chainA, e.Chains[chainC].Users[1])
+		assertRequestSent(chainB, chainC, e.Chains[chainB].Users[1])
+		assertRequestSent(chainC, chainB, e.Chains[chainC].Users[1])
+	}()
+	// disable lanes between A & C and C & B while requests are getting sent
+	testhelpers.RemoveLane(t, &tenv, chainA, chainC, false)
+	testhelpers.RemoveLane(t, &tenv, chainC, chainA, false)
+	testhelpers.RemoveLane(t, &tenv, chainB, chainC, false)
+	testhelpers.RemoveLane(t, &tenv, chainC, chainB, false)
```

Comment:
This could lead to flakiness I think. How about -
1. Just use one lane and send 10 requests
2. Remove that lane.
3. Ensure all requests are delivered even after removing lane